cache/models/samples.js

```diff
@@ -561,20 +582,15 @@ function upsertOneParsedSample(sampleQueryBodyObj, parsedSample, isBulk, user) {
     }
 
     // sampleQueryBodyObj updated with fields
-    updateSampleAttributes(sampleQueryBodyObj, sample, aspect);
-
+    return updateSampleAttributes(sampleQueryBodyObj, sample);
+  })
+  .then(() => {
     if (sample) { // if sample exists, just update sample
       delete sampleQueryBodyObj.name; // to avoid updating sample name
       logInvalidHmsetValues(sampleKey, sampleQueryBodyObj);
       return redisClient.hmsetAsync(sampleKey, sampleQueryBodyObj);
     }
 
-    /*
-     * Otherwise the sample is new. Set the name to be the combination of
-     * subject absolutePath and aspect name.
-     */
-    sampleQueryBodyObj.name = subject.absolutePath + '|' + aspect.name;
-
     // Add the provider and user fields.
```

Comment:
Not sure. Yes, the name is required in request body.
Probably because we have had some issues related to samples without a name and we are making sure that the sample has a name here?
Oh you are asking because... maybe we should keep this line? hmm, I think we don't need this.

Reply:
I agree, I don't think we need this. Just wanted to understand why it was there before, in case there was some reason we were missing. I think you're right, probably related to the samples-without-a-name issues in the past.